src/lightning/pytorch/utilities/compile.py

```diff
@@ -134,7 +134,7 @@ def _maybe_unwrap_optimized(model: object) -> "pl.LightningModule":
 def _verify_strategy_supports_compile(model: "pl.LightningModule", strategy: Strategy) -> None:
     if model._compiler_ctx is not None:
         supported_strategies = (SingleDeviceStrategy, DDPStrategy, FSDPStrategy)
-        if not isinstance(strategy, supported_strategies):
+        if not isinstance(strategy, supported_strategies) or isinstance(strategy, DeepSpeedStrategy):
```

Comment:
the DeepSpeedStrategy inherits from DDPStrategy, the previous instance check is not enough.